app/Http/Requests/AcceptSubmissionRequest.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Requests;
+
+use App\Enums\UserTypes;
+use Illuminate\Foundation\Http\FormRequest;
+
+class AcceptSubmissionRequest extends FormRequest
```

Comment:
Es innecesario este Form Request dado que estas haciendo manualmente lo que el paquete de Spatie ya provee, un middleware para validar el rol del usuario: https://spatie.be/docs/laravel-permission/v5/basic-usage/middleware

En cuanto a la validacion para si la submission ya esta tomada, podrias probar definir una resolucion custom para el route model binding, podes leer respecto a esto [aca](https://laravel.com/docs/10.x/routing#customizing-the-resolution-logic). En esencia podrias bindear para la key 'pendingSubmission', por ejemplo, que realice una query donde `doctor_id` sea `null`, y de esa manera podes usar a nivel global este tipo de binding y despreocuparte de validar si la submission esta tomada o no.